backend/social/models.py

```diff
@@ -27,3 +28,29 @@ def add_message(sender, instance, created, **kwargs):
     from social.subscriptions import MessageSubscription
     if created:
         MessageSubscription.broadcast(payload={'message': instance}, group=f'group_{instance.group.id}')
+
+#Comprobar que un usuario no invie un mensaje a un grupo al que no pertenece
+@receiver(pre_save, sender=Message)
+def check_user_in_group(sender, instance, **kwargs):
+    if instance.user not in instance.group.users.all():
+        raise Exception('User not in group')
+
+
+# Comrpobar que un grupo individual solo tenga un usuario
+@receiver(pre_save, sender=Group)
+def check_individual_group(sender, instance, **kwargs):
+    if instance.individual:
+        if instance.users.count() > 1:
```

Comment:
Unifica ambas condiciones en la siguiente: `if instance.individual and instance.users.count() != 2:`. Nótese que se cambia también la validación pues el número total de usuarios en un Group de tipo individual es siempre 2.
También haría falta añadir una restricción para no permitir la creación de grupos no individuales con menos de 3 usuarios.